x11-packages/wlroots/0006-fix-vulkan-renderer-build.patch

```diff
@@ -0,0 +1,277 @@
++	0x00050048,0x00000015,0x00000000,0x00000023,0x00000050,0x00030047,0x00000015,0x00000002,
++	0x00020013,0x00000002,0x00030021,0x00000003,0x00000002,0x00030016,0x00000006,0x00000020,
++	0x00040017,0x00000007,0x00000006,0x00000004,0x00040020,0x00000008,0x00000003,0x00000007,
++	0x0004003b,0x00000008,0x00000009,0x00000003,0x00090019,0x0000000a,0x00000006,0x00000001,
++	0x00000000,0x00000000,0x00000000,0x00000001,0x00000000,0x0003001b,0x0000000b,0x0000000a,
++	0x00040020,0x0000000c,0x00000000,0x0000000b,0x0004003b,0x0000000c,0x0000000d,0x00000000,
++	0x00040017,0x0000000f,0x00000006,0x00000002,0x00040020,0x00000010,0x00000001,0x0000000f,
++	0x0004003b,0x00000010,0x00000011,0x00000001,0x0004002b,0x00000006,0x00000013,0x00000000,
++	0x0003001e,0x00000015,0x00000006,0x00040020,0x00000016,0x00000009,0x00000015,0x0004003b,
++	0x00000016,0x00000017,0x00000009,0x00040015,0x00000018,0x00000020,0x00000001,0x0004002b,
++	0x00000018,0x00000019,0x00000000,0x00040020,0x0000001a,0x00000009,0x00000006,0x00050036,
++	0x00000002,0x00000004,0x00000000,0x00000003,0x000200f8,0x00000005,0x0004003d,0x0000000b,
++	0x0000000e,0x0000000d,0x0004003d,0x0000000f,0x00000012,0x00000011,0x00070058,0x00000007,
++	0x00000014,0x0000000e,0x00000012,0x00000002,0x00000013,0x0003003e,0x00000009,0x00000014,
++	0x00050041,0x0000001a,0x0000001b,0x00000017,0x00000019,0x0004003d,0x00000006,0x0000001c,
++	0x0000001b,0x0004003d,0x00000007,0x0000001d,0x00000009,0x0005008e,0x00000007,0x0000001e,
++	0x0000001d,0x0000001c,0x0003003e,0x00000009,0x0000001e,0x000100fd,0x00010038
++};
+diff --git a/render/vulkan/vulkan.c b/render/vulkan/vulkan.c
+index 61f49ea5..307eec81 100644
+--- a/render/vulkan/vulkan.c
++++ b/render/vulkan/vulkan.c
+@@ -26,6 +26,7 @@ static bool check_extension(const VkExtensionProperties *avail,
+ 	return false;
+ }
+ 
++#if !defined (__ANDROID__) && !defined (__TERMUX__)
+ static VkBool32 debug_callback(VkDebugUtilsMessageSeverityFlagBitsEXT severity,
+ 		VkDebugUtilsMessageTypeFlagsEXT type,
+ 		const VkDebugUtilsMessengerCallbackDataEXT *debug_data,
+@@ -74,7 +75,7 @@ static VkBool32 debug_callback(VkDebugUtilsMessageSeverityFlagBitsEXT severity,
+ 
+ 	return false;
+ }
+-
++#endif
+ 
+ // instance
+ struct wlr_vk_instance *vulkan_instance_create(bool debug) {
+@@ -126,12 +127,14 @@ struct wlr_vk_instance *vulkan_instance_create(bool debug) {
+ 	size_t extensions_len = 0;
+ 	const char *extensions[1] = {0};
+ 
++#if !defined (__ANDROID__) && !defined (__TERMUX__)
+ 	bool debug_utils_found = false;
+ 	if (debug && check_extension(avail_ext_props, avail_extc,
+ 			VK_EXT_DEBUG_UTILS_EXTENSION_NAME)) {
+ 		debug_utils_found = true;
+ 		extensions[extensions_len++] = VK_EXT_DEBUG_UTILS_EXTENSION_NAME;
+ 	}
++#endif
+ 
+ 	assert(extensions_len <= sizeof(extensions) / sizeof(extensions[0]));
+ 
+@@ -143,7 +146,9 @@ struct wlr_vk_instance *vulkan_instance_create(bool debug) {
+ 	};
+ 
+ 	const char *layers[] = {
++#if !defined (__ANDROID__) && !defined (__TERMUX__)
```

Comment:
Wouldn't it be easier to just patch the default_debug parameter at the call site of this function?

Also this layer array is already gone in wlroots 0.17, an upgrade could also help.
https://gitlab.freedesktop.org/wlroots/wlroots/-/blob/0.17/render/vulkan/vulkan.c?ref_type=heads#L151